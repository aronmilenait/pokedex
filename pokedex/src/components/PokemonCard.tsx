interface PokemonCardProps {
  pokemon: {
    name: string;
    id: number;
    image: string;
  };
}

const PokemonCard = ({ pokemon }: PokemonCardProps) => {

  return (
    <div className="bg-light-blue p-4 text-center shadow-md">
      <img
        src={pokemon.image}
        alt={pokemon.name}
        className="w-24 mx-auto mb-4 hover:scale-125 transition-transform duration-200 ease-in-out"
      />
      <p className="text-lg font-bold text-white text-shadow">{pokemon.name}</p>
    </div>
  );
};

export default PokemonCard;
