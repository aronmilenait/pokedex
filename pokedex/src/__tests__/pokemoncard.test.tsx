import { render } from '@testing-library/react';
import PokemonCard from '../components/PokemonCard';

test('renders PokemonCard', () => {
  const pokemon = {
    name: 'Vulpix',
    id: 37,
    image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/37.png',
  };

  const { container } = render(<PokemonCard pokemon={pokemon} />);
  expect(container).toMatchSnapshot();
});
