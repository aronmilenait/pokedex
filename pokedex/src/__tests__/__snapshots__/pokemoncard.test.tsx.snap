// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PokemonCard 1`] = `
<div>
  <div
    class="bg-light-blue p-4 text-center shadow-md"
  >
    <img
      alt="Vulpix"
      class="w-24 mx-auto mb-4 hover:scale-125 transition-transform duration-200 ease-in-out"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/37.png"
    />
    <p
      class="text-lg font-bold text-white text-shadow"
    >
      Vulpix
    </p>
  </div>
</div>
`;
